From 9ecbe0eba25b0f1fc00370f6ffd9b901faf05575 Mon Sep 17 00:00:00 2001
From: Syed Hasan Raza Naqvi <syed.naqvi@broadcom.com>
Date: Wed, 4 May 2022 05:11:01 +0000
Subject: [PATCH] pceplib: disable IPv6 source address bind in SONiC test

The SONiC build container loopback does not enable IPv6. Use the unspecified
IPv6 source address so the socket test does not fail with EADDRNOTAVAIL.

Original-patch-commit: dfe0158344ea581370dd2dba8e4db189cf369147
---
 pceplib/test/pcep_socket_comm_test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pceplib/test/pcep_socket_comm_test.c b/pceplib/test/pcep_socket_comm_test.c
index c0b73c3dbc..4a70e49fb7 100644
--- a/pceplib/test/pcep_socket_comm_test.c
+++ b/pceplib/test/pcep_socket_comm_test.c
@@ -80,7 +80,7 @@ void pcep_socket_comm_test_setup()
 	inet_pton(AF_INET, "127.0.0.1", &(test_host_ip));
 	inet_pton(AF_INET, "127.0.0.1", &(test_src_ip));
 	inet_pton(AF_INET6, "::1", &(test_host_ipv6));
-	inet_pton(AF_INET6, "::1", &(test_src_ipv6));
+	inet_pton(AF_INET6, "::", &(test_src_ipv6));
 }
 
 void pcep_socket_comm_test_teardown()
-- 
2.34.1

